### PR TITLE
Accept headers

### PR DESCRIPTION
**Pull request by sduckett**
This PR is a sibling of [a notification-works PR](https://github.com/democracyworks/notification-works/pull/41) that gets headers into a notification type. The sibling PR would send a map of headers to email-works, which, due to some Postal-details, needs those headers flattened into the message. Here, we accept a map and merge its (scalar) contents before transmitting the message.

[The Story](https://www.pivotaltracker.com/story/show/151441503).

### CODE REVIEW

File: test/email_works/handlers_test.clj [hunk 1]
```diff
@@ -26,6 +26,23 @@
                               :type :attachment}))}
                 (:body email))))))
 
+(deftest fold-headers-test
+    (let [m {:to "recipient@example.com"
+             :from "sender@example.com"
+             :subject "a message"
+             :body "may contain interesting content"}
+          flat {"x-magic-powers" "hiterto-unseen"}
```
Comment (gws) [inline test/email_works/handlers_test.clj]: `hitherto`

:trollface:

File: test/email_works/handlers_test.clj [hunk 2]
```diff
@@ -26,6 +26,23 @@
...
+          nested {"scalar" "swift"
+                  "nested" {"nefarious" "nematoads"}}]
+      (testing "optional `:headers` are merged into the message"
+        (is (= #{:to :from :subject :body}
+               (-> m fold-headers keys set)))
+        (is (= #{:to :from :subject :body "x-magic-powers"}
+               (-> (assoc m :headers flat) fold-headers keys set))))
+      (testing "any nested collections are removed"
+        (is (= #{:to :from :subject :body "scalar"}
+               (-> (assoc m :headers nested) fold-headers keys set))) )))
```
Comment (gws) [inline test/email_works/handlers_test.clj]: extra space

Comment (sduckett) [reply test/email_works/handlers_test.clj]: @gws, I'm picking you for my _Where's Waldo?_ team.

File: src/email_works/handlers.clj [hunk 3]
```diff
@@ -71,6 +71,17 @@
         (update :body (comp vec concat) (map attach-file attachments)))
     email))
 
+(defn fold-headers
+  "If a `:headers` map is present in the message, merge it to the message.
+  Nested collections will be filtered out."
+  [m]
+  (if-let [hs (->> (:headers m)
+                   (filter (fn [[k v]] (not (coll? v))))
```
Comment (gws) [inline src/email_works/handlers.clj]: I think we could refactor this to use `remove` and get rid of the `not`

**Comment by gws** [comment]
OK that's my last comment. LGTM!

File: src/email_works/handlers.clj (hunk 3 above)
Comment (sduckett) [reply src/email_works/handlers.clj]: Fair point; my brain seems to prefer the verbose form, so keep your eyes open!

**Pull request merged**